### PR TITLE
Cloudsnap fork

### PR DESCRIPTION
**Pull request by danielpowell4**
### What does this PR do?

- Changes ruby support to current stable per https://www.ruby-lang.org/en/downloads/
- Adds legacy support per https://github.com/CloudsnapInc/intacct-ruby/pull/1
- Allows for locationid in header per https://github.com/CloudsnapInc/intacct-ruby/pull/2
- Provides for ordering in legacy requests per https://github.com/CloudsnapInc/intacct-ruby/pull/4
- Optional keys in legacy endpoints per https://github.com/CloudsnapInc/intacct-ruby/pull/6

### How do I manually test this?

Review tests, trust CI!

### Additional Comments

Big thanks to @reginad1 @draycox and @lifeBCE!

### CODE REVIEW

**Comment by danielpowell4** [comment]
@jakemaas1 / @toddsiegel making tweaks. Will force push a cleaner commit history with updates to changelog soon!

File: lib/intacct_ruby/request.rb [hunk 1]
```diff
@@ -120,6 +120,7 @@ def authentication_block
           @request.userid    @opts[:userid]
           @request.companyid @opts[:companyid]
           @request.password  @opts[:user_password]
+          @request.locationid @opts[:locationid]
```
Comment (danielpowell4) [inline lib/intacct_ruby/request.rb]: @jakemaas1 do you think this will cause us an issue--mostly with VCRs

Should we perhaps include a guard?

**Comment by Timothyjb** [comment]
Hi @danielpowell4, I know this one has been open for a while. Is there anything that any of us at Cloudsnap could do to help this move forward? 
We've made some other updates after these that could be useful to others as well. If it is simpler/easier I can open them up in smaller PRs.

**Comment by jakemaas1** [comment]
Hi @Timothyjb, I know it has been years since any movement here, but just wanted to let you know that our team has decided to internalize the parts of this gem we need and archive the repo going forward. 

Please let me know if there's something we can do before archival to make it easier for you all - it looks like your fork may be ahead of this anyway by now?

**Comment by Timothyjb** [comment]
@jakemaas1 Hi, thanks for the heads up! Nothing from us, we are all set.

**Pull request closed without merging**